File: FastKV/src/main/java/io/fastkv/LongListEncoder.java
package io.fastkv;

import io.packable.PackDecoder;
import io.packable.PackEncoder;

import java.util.ArrayList;
import java.util.List;

public class LongListEncoder implements FastKV.Encoder<List<Long>> {
    public static final LongListEncoder INSTANCE = new LongListEncoder();

    private LongListEncoder() {
    }

    @Override
    public String tag() {
        return "LongList";
    }

    @Override
    public byte[] encode(List<Long> obj) {
        return new PackEncoder().putLongList(0, obj).getBytes();
    }

    @Override
    public List<Long> decode(byte[] bytes, int offset, int length) {
        return PackDecoder.newInstance(bytes, offset, length).getLongList(0);
    }
}

File: FastKVDemo/app/src/main/java/io/fastkv/fastkvdemo/fastkv/LongListEncoder.kt
package io.fastkv.fastkvdemo.fastkv

import io.fastkv.FastKV
import io.packable.PackDecoder
import io.packable.PackEncoder

object LongListEncoder : FastKV.Encoder<List<Long>> {
    override fun tag(): String {
        return "LongList"
    }

    override fun encode(obj: List<Long>): ByteArray {
        return PackEncoder().putLongList(0, obj).bytes
    }

    override fun decode(bytes: ByteArray, offset: Int, length: Int): List<Long>? {
        return PackDecoder.newInstance(bytes, offset, length).getLongList(0)
    }
}